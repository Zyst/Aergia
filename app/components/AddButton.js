// @flow
import React, { Component } from "react";
import Mousetrap from "mousetrap";
import hotkeys from "../utils/hotkeys";

type Props = {
  add: () => void
};

class AddButton extends Component {
  props: Props;

  componentDidMount() {
    Mousetrap.bind(hotkeys.pomodoro.add.addButton.callAdd.key, e => {
      e.preventDefault();
      this.props.add();
    });
  }

  componentWillUnmount() {
    Mousetrap.unbind(hotkeys.pomodoro.add.addButton.callAdd.key);
  }

  render() {
    return (
      <button className="center-block button-primary" onClick={this.props.add}>
        Add task
      </button>
    );
  }
}

export default AddButton;
