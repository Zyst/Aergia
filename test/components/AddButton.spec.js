import { spy } from "sinon";
import React from "react";
import { shallow } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import AddButton from "../../app/components/AddButton";

describe("Add Button component", () => {
  it("should be defined", () => {
    expect(AddButton).toBeDefined();
  });
});
