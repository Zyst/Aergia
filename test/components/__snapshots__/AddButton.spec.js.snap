// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Button component Should match snapshot 1`] = `
<button
  className="center-block button-primary"
  onClick={[Function]}
>
  Add task
</button>
`;
